File: WOT-iOS/WOT-iOS/core/datasources/WOTTree/WOTTree+Pivot/WOTPivotFilterNode.m
//
//  WOTPivotFilterNode.m
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 8/27/15.
//  Copyright (c) 2015 Pavel Yeshchyk. All rights reserved.
//

#import "WOTPivotFilterNode.h"

@implementation WOTPivotFilterNode

- (NSInteger)y {
    
    return 0;
}

- (NSInteger)x {
    
    return 0;
}

- (NSInteger)width {
    
    return self.dimensionDelegate.rootNodeWidth;
}

- (NSInteger)height {
    
    return self.dimensionDelegate.rootNodeHeight;
}

- (PivotStickyType)stickyType {
    
    return PivotStickyTypeHorizontal | PivotStickyTypeVertical;
}

@end

File: WOT-iOS/WOT-iOS/core/datasources/WOTTree/WOTTree+Pivot/WOTPivotFilterNode.swift
//
//  WOTPivotFilterNode.swift
//  WOT-iOS
//
//  Created by Pavel Yeshchyk on 7/17/18.
//  Copyright © 2018 Pavel Yeshchyk. All rights reserved.
//

import Foundation

class WOTPivotFilterNodeSwift: WOTPivotNodeSwift {

    override var stickyType: PivotStickyType {
//        return PivotStickyTypeHorizontal | PivotStickyTypeVertical;
        return (PivotStickyType(rawValue: PivotStickyType.RawValue(UInt8(PivotStickyType.horizontal.rawValue) | UInt8(PivotStickyType.vertical.rawValue))))!
    }
}
